src/SpectralDataManager.cpp

```diff
@@ -0,0 +1,214 @@
+#include <iostream>
+#include "FFT.h"
+#include "SpectralDataManager.h"
+#include "WaveTrack.h"
+
+SpectralDataManager::SpectralDataManager()= default;
+
+SpectralDataManager::~SpectralDataManager()= default;
+
+struct SpectralDataManager::Setting{
+   eWindowFunctions mInWindowType = eWinFuncHann;
+   eWindowFunctions mOutWindowType = eWinFuncHann;
+   size_t mWindowSize = 2048;
+   unsigned mStepsPerWindow = 4;
+   bool mLeadingPadding = true;
+   bool mTrailingPadding = true;
+};
+
+int SpectralDataManager::ProcessTracks(TrackList &tracks){
+   int applyCount = 0;
+   Setting setting;
+   Worker worker(setting);
+
+   for ( auto wt : tracks.Any< WaveTrack >() ) {
+      auto &trackView = TrackView::Get(*wt);
+
+      if(auto waveTrackViewPtr = dynamic_cast<WaveTrackView*>(&trackView)){
+         for(const auto &subViewPtr : waveTrackViewPtr->GetAllSubViews()){
+            if(!subViewPtr->IsSpectral())
+               continue;
+            auto sView = std::static_pointer_cast<SpectrumView>(subViewPtr).get();
+            auto pSpectralData = sView->GetSpectralData();
+
+            if(!pSpectralData->dataHistory.empty()){
+               worker.Process(wt, pSpectralData);
+               applyCount += static_cast<int>(pSpectralData->dataHistory.size());
+               pSpectralData->clearAllData();
+            }
+         }
+      }
+   }
+
+   return applyCount;
+}
+
+wxInt64 SpectralDataManager::FindFrequencySnappingBin(WaveTrack *wt,
+                                                      long long int startSC,
+                                                      double threshold,
+                                                      wxInt64 targetFreq)
+{
+   Setting setting;
+   Worker worker(setting);
+
+   return worker.ProcessSnapping(wt, startSC, threshold, targetFreq);
+}
+
+SpectralDataManager::Worker::Worker(const Setting &setting)
+:TrackSpectrumTransformer{ true, setting.mInWindowType, setting.mOutWindowType,
```

Comment:
I see a problem.  You always pass true as the first argument to the base class constructor, which means output is required, the inverse Fourier transform is done, and overlap-add, and pasting of data into the track that was supplied to Process().

All of this is unnecessary when you are only computing frequency snapping.  Could this be another reason for the slowness?